### DISCUSSION TITLE
Best way to initialize data from React Query (using tRPC) into local state

### DISCUSSION THREAD

**Discussion by vaunblu** [pmndrs/jotai#1557] [Q&A]
I hope this is less of a philosophical question, but I am trying to create an app that pulls data from my SQL database, allows one to rapidly change the state client side, and then post the updates back to the database when a user hits save. 

I found Jotai is the easiest way to alter the state client side, but I can't seem initialize the atom outside of useEffect, but then I also can't seem to not overwrite my data in that same useEffect hook. 

```javascript
const data = trpc.getData.useQuery()

useEffect(() => {
  if (data.isSuccess && data.data) {
    setJotaiState(data.data);
  }
}, [data, setJotaiState]);
```
I have some other dependencies in the useEffect so whenever it triggers, my data resets. 

I hope this is not a super noob question.

**Comment by dai-shi** [comment]
## Re: usage of useEffect

You can define another useEffect with dependencies you want.

## Re: write atoms from props

atoms don't belong to React, so, using `useEffect` is the right solution.
For filling the initial atom values (only the first time), `useHydrateAtoms` could do.

## Re: Jotai with tRPC

Check out: https://github.com/jotai-labs/jotai-trpc/pull/6
